packages/catalog-model/src/location/types.ts

```diff
@@ -17,6 +17,9 @@
 export type LocationSpec = {
   type: string;
   target: string;
+  // When using repo importer plugin, location is being created before the component yaml file is merged to the main branch.
+  // This flag is then set to disable validation that prevents creation of location if target file does not yet exist.
+  pendingLocation?: boolean;
```

Comment:
Hmmmm catalog processors currently use the syntax "optional" for this https://github.com/spotify/backstage/blob/master/plugins/catalog-backend/src/ingestion/processors/types.ts#L30

Suggested change: `presence?: 'optional' | 'required'`, and document that its default value is `'required'`.

That way, I will be able to make use of this field in refactoring the processors accordingly later, and even store that new field in the database because later runs of the refresh loop should know not to log warnings when the location isn't there

Reply:
Done.

Reply:
Done.